Add environment variables reference to cli-ref.rst (#10970)
mistercrunch/Airflow: docs/cli-ref.rst

   :module: airflow.cli.cli_parser
   :func: get_parser
   :prog: airflow

Environment Variables
'''''''''''''''''''''

.. envvar:: AIRFLOW__{SECTION}__{KEY}

  Any specific key in a section in Airflow.

>   The user may replace the {SECTION} placeholder with any section

Reviewer: ```suggestion
  The user may replace the ``{SECTION}`` placeholder with any section
```

>   and the {KEY} placeholder with any key in that specified section.

Reviewer: ```suggestion
  and the ``{KEY}`` placeholder with any key in that specified section.
```

  For example, the user may initiate "AIRFLOW__CORE__DAGS_FOLDER",
  where CORE is the section and DAGS_FOLDER is the key within the section.

>   If a {SECTION} - {KEY} pair exists in the configuration file, it will be overwritten.

Reviewer: ```suggestion
  For more information, see: :doc:`/howto/set-config`.
```

.. envvar:: AIRFLOW__{SECTION}__{KEY}_CMD

  For any specific key in a section in Airflow,
  execute the command the key is pointing to.

.. envvar:: AIRFLOW_CONFIG

  The path to the Airflow configuration file.

.. envvar:: AIRFLOW_CONN_{CONN_ID}

  Defines a connection URI.

>   The user may replace the {CONN_ID} placeholder with the connection name.

Reviewer: ```suggestion
  The user may replace the ``{CONN_ID}`` placeholder with the connection name.
```

  For example, the user may create a connection named "AIRFLOW_CONN_PROXY_POSTGRES_TCP"

>   to connect via proxy over TCP.

Reviewer: ```suggestion
  to connect via proxy over TCP.
  
  For more information, see: :ref:`environment_variables_secrets_backend`.
```

.. envvar:: AIRFLOW_HOME

  The root directory for the Airflow content.
  This is the default parent directory for Airflow assets such as DAGs and logs.

.. envvar:: AIRFLOW_VAR_{KEY}

  Defines an Airflow variable.

>   The user may replace the {KEY} placeholder with the variable name.

Reviewer: ```suggestion
  The user may replace the ``{KEY}`` placeholder with the variable name.
  
  For more information, see: <Link to Storing Variables in Environment Variables on https://airflow.readthedocs.io/en/latest/howto/variable.html>
```

   :module: airflow.cli.cli_parser
   :func: get_parser
   :prog: airflow

Environment Variables
'''''''''''''''''''''

.. envvar:: AIRFLOW__{SECTION}__{KEY}

>   Any specific key in a section in Airflow.

Reviewer: This sentence does not explain what this variable does. The word "configuration" is at least missing.

> Sets options in the Airflow configuration. This takes priority over the value in the ``airflow.cfg`` file.

  The user may replace the ``{SECTION}`` placeholder with any section
  and the ``{KEY}`` placeholder with any key in that specified section.

>   For example, the user may initiate "AIRFLOW__CORE__DAGS_FOLDER",
>   where CORE is the section and DAGS_FOLDER is the key within the section.

Reviewer: ```suggestion
  For example, If you want to set the ``dags_folder`` options in ``[core]`` section, then you 
   should set the ``AIRFLOW__CORE__DAGS_FOLDER``  environment variable.
```
There are a few problems here. First, not the best tone of speech. The word may in context suggests that there is another variable name as well. Second, all sentences should address the reader directly. Third,  if you refer to a section name it should always be written in square brackets as a code literal.

Reviewer: I hope you do not offend my comments, but I do try to make the documentation accessible to a wide audience.
I believe that every developer should read Google's guide, which highlights some good practices.
https://developers.google.com/style

  For more information, see: :doc:`/howto/set-config`.

.. envvar:: AIRFLOW__{SECTION}__{KEY}_CMD

  For any specific key in a section in Airflow,
  execute the command the key is pointing to.

.. envvar:: AIRFLOW_CONFIG

  The path to the Airflow configuration file.

.. envvar:: AIRFLOW_CONN_{CONN_ID}

>   Defines a connection URI.

Reviewer: This does not define a connection URI, but a new connection using the URI.

>   The user may replace the ``{CONN_ID}`` placeholder with the connection name.

Reviewer: ```suggestion
  You may replace the ``{CONN_ID}`` placeholder with the connection name.
```

  For example, the user may create a connection named "AIRFLOW_CONN_PROXY_POSTGRES_TCP"
  to connect via proxy over TCP.

  For more information, see: :ref:`environment_variables_secrets_backend`.

.. envvar:: AIRFLOW_HOME

  The root directory for the Airflow content.
  This is the default parent directory for Airflow assets such as DAGs and logs.

.. envvar:: AIRFLOW_VAR_{KEY}

  Defines an Airflow variable.

>   The user may replace the ``{KEY}`` placeholder with the variable name.

Reviewer: ```suggestion
  You may replace the ``{KEY}`` placeholder with the variable name.
```
I think we should address the reader directly. Use you, instead of the user or they. 
See: https://developers.google.com/style/person

>   For more information, see: <Link to Storing Variables in Environment Variables on https://airflow.readthedocs.io/en/latest/howto/variable.html>

Reviewer 2: ```suggestion
  For more information, see: <link to Storing Variables in Environment Variables on https://airflow.readthedocs.io/en/latest/howto/variable.html>.
```

>   For more information, see: <link to Storing Variables in Environment Variables on https://airflow.readthedocs.io/en/latest/howto/variable.html>.

Reviewer: Can you add use ref here? See: https://stackoverflow.com/questions/15394347/adding-a-cross-reference-to-a-subheading-or-anchor-in-another-page

Author: @mik-laj yes 😄 this is done!

   :module: airflow.cli.cli_parser
   :func: get_parser
   :prog: airflow

> Environment Variables

Reviewer 3: Curious why this is placed in cli-ref.rst -- any particular reason ?

Reviewer: It is common practice to include a list of the environment variables supported next to a command list. It makes sense from the user's perspective. If the user executes commands, they also has shell access to set environment variables.
Examplle: 
https://www.sphinx-doc.org/en/master/man/sphinx-build.html#environment-variables
https://docs.python.org/3/using/cmdline.html

Reviewer 3: We should change the title from Cli Ref to Cli Ref and Environment Variables like that Python docs link

'''''''''''''''''''''

.. envvar:: AIRFLOW__{SECTION}__{KEY}

  Sets options in the Airflow configuration. This takes priority over the value in the ``airflow.cfg`` file.

  Replace the ``{SECTION}`` placeholder with any section
  and the ``{KEY}`` placeholder with any key in that specified section.

  For example, if you want to set the ``dags_folder`` options in ``[core]`` section,
  then you should set the ``AIRFLOW__CORE__DAGS_FOLDER`` environment variable.

  For more information, see: :doc:`/howto/set-config`.

.. envvar:: AIRFLOW__{SECTION}__{KEY}_CMD

>   For any specific key in a section in Airflow,

Reviewer: This section does not say that this is used to set configuration options. It only says that the command is run, but not why it is run. It is worth writing that the command is run and the result is used as a value for the configuration option.
